src/scikit_build_core/builder/builder.py

```diff
@@ -123,19 +187,29 @@ def configure(
         }
 
         # Add any extra CMake modules
-        eps = metadata.entry_points(group="cmake.module")
-        self.config.module_dirs.extend(resources.files(ep.load()) for ep in eps)
+        _handle_search_paths(
+            "cmake.module", self.settings.search.modules, self.config.module_dirs
+        )
 
         # Add any extra CMake prefixes
-        eps = metadata.entry_points(group="cmake.prefix")
-        self.config.prefix_dirs.extend(resources.files(ep.load()) for ep in eps)
+        _handle_search_paths(
+            "cmake.prefix", self.settings.search.prefixes, self.config.prefix_dirs
+        )
+
+        # Add all CMake roots
+        # TODO: Check for unique uppercase names
+        _handle_search_paths(
+            "cmake.root", self.settings.search.roots, self.config.prefix_roots
+        )
 
         # Add site-packages to the prefix path for CMake
         site_packages = Path(sysconfig.get_path("purelib"))
-        self.config.prefix_dirs.append(site_packages)
+        if self.settings.search.use_install_prefix:
```

Comment:
As I said above, that's not what this is. We are trying to add site-packages, and in some cases, that's not where scikit-build-core is installed, so we make sure we always add that too. We should probably be adding platlib too.

Reply:
Hmm, so each isolated build path is unique to the packages being built and not to the call of `pip install`? I was mostly considering about that situation where you would need the prefixes of dependencies that are in `build-system.requires` only.

Reply:
You cannot view any other site-packages besides the isolated one (that's the isolated part!). The wheel is built separately from installing it. This was just a workaround for an issue where "site-packages" wasn't pointing to the place scikit-build-core was installed, I think due to user site-packages for non-isolated installs, but I don't remember exactly. It could also have been the `include-system-site-packages = true` setting for a (non-isolated) venv. But it's not install vs. build.

Reply:
> You cannot view any other site-packages besides the isolated one (that's the isolated part!).

Wait, but we saw that breaking in #860 where the installing site-packages were included. I'll run the build for `spglib` again to confirm that it was indeed that case.

Reply:
Indeed this is what I got:
```console
$ pip -v install ./spglib --config-settings=logging.level=DEBUG

  2024-09-09 10:41:40,787 - scikit_build_core - DEBUG - SITE_PACKAGES: /home/lecris/CLionProjects/spglib/venv/lib/python3.12/site-packages
  2024-09-09 10:41:40,787 - scikit_build_core - DEBUG - Extra SITE_PACKAGES: /home/lecris/CLionProjects/spglib/venv/lib/python3.12/site-packages
  2024-09-09 10:41:40,798 - scikit_build_core - DEBUG - Default generator: Ninja

    set(CMAKE_PREFIX_PATH [===[/home/lecris/CLionProjects/spglib/venv/lib/python3.12/site-packages;/tmp/pip-build-env-b9zh9nkx/overlay/lib/python3.12/site-packages]===] CACHE PATH "" FORCE)
```
note the actual value is in `CMAKE_PREFIX_PATH` due to a minor bug:
https://github.com/scikit-build/scikit-build-core/blob/9f544439ee2a9109ca6fd2b4af2a37f663c5d177/src/scikit_build_core/builder/builder.py#L138-L139

In this case we have `/tmp/pip-build-env-` site-packages which points to where `scikit-build-core` was installed, which in the issolated build it would be this "build environment". And what I find is that the `numpy` is also installed in the same paths, and this is mostly what I was thinking of making configurable.

In contrast if I run it without build-isolation:
```console
$ pip install scikit-build-core numpy
$ pip -v install ./spglib --config-settings=logging.level=DEBUG --no-build-isolation

  2024-09-09 10:53:58,703 - scikit_build_core - DEBUG - SITE_PACKAGES: /home/lecris/CLionProjects/spglib/venv/lib/python3.12/site-packages
  2024-09-09 10:53:58,703 - scikit_build_core - DEBUG - FindPython backport activated at /home/lecris/CLionProjects/spglib/venv/lib/python3.12/site-packages/scikit_build_core/resources/find_python
  2024-09-09 10:53:58,706 - scikit_build_core - DEBUG - Default generator: Ninja

    set(CMAKE_PREFIX_PATH [===[/home/lecris/CLionProjects/spglib/venv/lib/python3.12/site-packages]===] CACHE PATH "" FORCE)
```

Of course `site_packages != DIR.parent.parent` is not a good check for if it's in the build-isolation because it intersect with `scikit-build-core` being editable installed, but in that case we know for sure we are running under `--no-build-isolation` and the other build dependencies are either in `site_packages` or their own editable path, which we do not have a way to control without them exporting their entry-point, so it technically still does what it "should do" similar to if `scikit-build-core` is not editable installed but `--no-build-isolation` is still used. 

|                                                        | `site_packages != DIR.parent.parent` |     `DIR.parent.parent`    | `use-build-prefix` |
|--------------------------------------------------------|:------------------------------------:|:--------------------------:|:------------------:|
| `scikit-build-core` in isolation                       |                `True`                | `/tmp/pip-build-env-*/...` |    "as intended"   |
| `pip install scikit-build-core` (without isolation)    |                `False`               |         `venv/...`         |       ignored      |
| `pip install -e scikit-build-core` (without isolation) |                `True`                |  `~/scikit-build-core/src` |      edge case     |

Reply:
Strange, why is `Path(sysconfig.get_path("purelib"))` reporting the original venv when this is "isolated"? It seems to me it should report the isolated environment and you should not be able to get to the original one at all. Is that path in `sys.path`?

Reply:
Welp, how would I test that? It is not reported in the debug log, and if I would run it within the `venv` than yeah it should be included right?
```
$ source venv/bin/activate && python3 -c "import sys;print(sys.path)"
['', '/usr/lib64/python312.zip', '/usr/lib64/python3.12', '/usr/lib64/python3.12/lib-dynload', '/home/lecris/CLionProjects/spglib/venv/lib64/python3.12/site-packages', '/home/lecris/CLionProjects/spglib/venv/lib/python3.12/site-packages']
```
Probably when running with isolation it would run in a different created environment in which case it would need to call `sys.path` within the build steps to show its true value right?

Reply:
It will be reported in the debug log after #896. :)

(And yes, sys.path inside the build steps is needed)

Reply:
Here's an update on it:
```
  2024-09-11 11:52:04,356 - scikit_build_core - DEBUG - SITE_PACKAGES: /home/lecris/CLionProjects/spglib/venv/lib/python3.12/site-packages
  2024-09-11 11:52:04,357 - scikit_build_core - DEBUG - Extra SITE_PACKAGES: /tmp/pip-build-env-oiycakwm/overlay/lib/python3.12/site-packages
  2024-09-11 11:52:04,357 - scikit_build_core - DEBUG - PATH: ['/home/lecris/CLionProjects/spglib/venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process', '/tmp/pip-build-env-oiycakwm/site', '/usr/lib64/python312.zip', '/usr/lib64/python3.12', '/usr/lib64/python3.12/lib-dynload', '/tmp/pip-build-env-oiycakwm/overlay/lib/python3.12/site-packages', '/tmp/pip-build-env-oiycakwm/overlay/lib64/python3.12/site-packages', '/tmp/pip-build-env-oiycakwm/normal/lib/python3.12/site-packages', '/tmp/pip-build-env-oiycakwm/normal/lib64/python3.12/site-packages', '/tmp/pip-build-env-oiycakwm/normal/lib/python3.12/site-packages/setuptools/_vendor']
```
The only path that is in the same venv is from `pip/_vendor/pyproject_hooks/_in_process`

Reply:
So this one is probably a `pip` bug. If I run `python3 -m build` or `uv pip install` then `platlib` is pointing to the isolated site-packages.